### PR TITLE
Yamas config written by node-agent

### PR DESCRIPTION
**Pull request by smorgrav**
These are all the relevant yamas checks that we identified last sprint.

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 1]
```diff
@@ -63,44 +66,94 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+        // ntp
+        Path ntpCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_ntp");
+        SecretAgentCheckConfig ntpSchedule = new SecretAgentCheckConfig("host-life", 60, ntpCheckPath);
+        configs.add(annotatedCheck(node, ntpSchedule));
+
+        // coredumps (except for the done coredumps which is handled by the host)
+        Path coredumpCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_coredumps");
+        SecretAgentCheckConfig coredumpSchedule = new SecretAgentCheckConfig("system-coredumps-processing", 300,
+                coredumpCheckPath, "--application", "system-coredumps-processing", "--lastmin",
+                "129600", "--crit", "1", "--coredir", "/home/y/var/crash/processing");
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `/home/y` -> `environment.pathInNodeUnderVespaHome()`

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 2]
```diff
@@ -63,44 +66,94 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+        List<SecretAgentCheckConfig> configs = new ArrayList<>();
 
+        // vespa
         Path vespaCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_vespa");
-        SecretAgentCheckConfig vespaSchedule = new SecretAgentCheckConfig("vespa", 60, vespaCheckPath, "all")
-                .withTag("parentHostname", environment.getParentHostHostname());
+        SecretAgentCheckConfig vespaSchedule = new SecretAgentCheckConfig("vespa", 60, vespaCheckPath, "all");
+        configs.add(annotatedCheck(node, vespaSchedule));
 
+        // vespa-health
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `yms_check_vespa` and `yms_check_host_life` do not work for config server

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 3]
```diff
@@ -63,44 +66,94 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+            // configserver
+            Path configServerCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_ymonsb2");
+            SecretAgentCheckConfig configServerSchedule = new SecretAgentCheckConfig("configserver", 60,
+                    configServerCheckPath, "-zero", "configserver");
+            configs.add(annotatedCheck(node, configServerSchedule));
+
+            //zkbackupage
+            Path zkbackupCheckPath = environment.pathInNodeUnderVespaHome("libexec/yamas2/yms_check_file_age.py");
+            SecretAgentCheckConfig zkbackupSchedule = new SecretAgentCheckConfig("zkbackupage", 300,
+                    zkbackupCheckPath, "-f", "/home/y/var/vespa-hosted/zkbackup.stat", "-m", "150", "-a", "config-zkbackupage");
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `/home/y` -> `environment.pathInNodeUnderVespaHome()`

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 4]
```diff
@@ -63,44 +66,94 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+            configs.add(annotatedCheck(node, zkbackupSchedule));
+        }
+
+        if (node.getNodeType() == NodeType.proxy) {
+            //routing-configage
+            Path routingAgeCheckPath = environment.pathInNodeUnderVespaHome("libexec/yamas2/yms_check_file_age.py");
+            SecretAgentCheckConfig routingAgeSchedule = new SecretAgentCheckConfig("routing-configage", 60,
+                    routingAgeCheckPath, "-f", "home/y/var/vespa-hosted/routing/nginx.conf", "-m", "90", "-a", "routing-configage");
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `/home/y` -> `environment.pathInNodeUnderVespaHome()`

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 5]
```diff
@@ -63,44 +66,94 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+        // host-life
         Path hostLifeCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_host_life");
-        SecretAgentCheckConfig hostLifeSchedule = new SecretAgentCheckConfig("host-life", 60, hostLifeCheckPath)
-                .withTag("namespace", "Vespa")
+        SecretAgentCheckConfig hostLifeSchedule = new SecretAgentCheckConfig("host-life", 60, hostLifeCheckPath);
+        configs.add(annotatedCheck(node, hostLifeSchedule));
+
+        // ntp
+        Path ntpCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_ntp");
+        SecretAgentCheckConfig ntpSchedule = new SecretAgentCheckConfig("host-life", 60, ntpCheckPath);
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `host-life` -> `ntp`

**Comment by freva** [comment]
Check failed in node-admin:
```
[ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.089 s <<< FAILURE! - in com.yahoo.vespa.hosted.node.admin.maintenance.StorageMaintainerTest
[ERROR] testMaintenanceThrottlingAfterSuccessfulMaintenance(com.yahoo.vespa.hosted.node.admin.maintenance.StorageMaintainerTest)  Time elapsed: 0.039 s  <<< ERROR!
java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /host/home/docker/container-storage/processed-coredumps
	at com.yahoo.vespa.hosted.node.admin.maintenance.StorageMaintainerTest.testMaintenanceThrottlingAfterSuccessfulMaintenance(StorageMaintainerTest.java:95)
Caused by: java.nio.file.NoSuchFileException: /host/home/docker/container-storage/processed-coredumps
	at com.yahoo.vespa.hosted.node.admin.maintenance.StorageMaintainerTest.testMaintenanceThrottlingAfterSuccessfulMaintenance(StorageMaintainerTest.java:95)
```

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java (hunk 1 above)
Comment (smorgrav) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: The check is ran in the container right?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java (hunk 2 above)
Comment (smorgrav) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: Do you know why host_life fails on the configserver?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java (hunk 1 above)
Comment (smorgrav) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: My underastanding was that pathInNodeUnderVespaHome is used to find the path when on the host but when you are in the container you would refer to the path directly (at least not the same path as when at the host)

**Comment by smorgrav** [comment]
@freva Think I covered your findings

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 6]
```diff
@@ -63,44 +66,98 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+        Path coredumpCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_coredumps");
+        SecretAgentCheckConfig coredumpSchedule = new SecretAgentCheckConfig("system-coredumps-processing", 300,
+                coredumpCheckPath, "--application", "system-coredumps-processing", "--lastmin",
+                "129600", "--crit", "1", "--coredir", environment.pathInNodeUnderVespaHome("var/crash/processing").toString());
+        configs.add(annotatedCheck(node, coredumpSchedule));
+
+        if (node.getNodeType() != NodeType.config) {
+            // vespa-health
+            Path vespaHealthCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_vespa_health");
+            SecretAgentCheckConfig vespaHealthSchedule = new SecretAgentCheckConfig("vespa", 60, vespaHealthCheckPath, "all");
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `vespa-health`

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java [hunk 7]
```diff
@@ -63,44 +66,98 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
...
+            //routing-configage
+            Path routingAgeCheckPath = environment.pathInNodeUnderVespaHome("libexec/yamas2/yms_check_file_age.py");
+            SecretAgentCheckConfig routingAgeSchedule = new SecretAgentCheckConfig("routing-configage", 60,
+                    routingAgeCheckPath, "-f", environment.pathInNodeUnderVespaHome("var/vespa-hosted/routing/nginx.conf").toString(),
+                    "-m", "90", "-a", "routing-configage");
+            configs.add(annotatedCheck(node, routingAgeSchedule));
+
+            //ssl-check
+            Path sslCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_ssl_status");
+            SecretAgentCheckConfig sslSchedule = new SecretAgentCheckConfig("ssh-status", 300,
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java]: `ssl-status`

**Review by freva — APPROVED** [review]
Minor comments, otherwise LGTM.

**Review by freva — APPROVED** [review]

**Pull request merged**